packages/platforms/dni/x86-64/x86-64-dni-wb2448/onlp/builds/src/module/src/sfpi.c

```diff
@@ -0,0 +1,160 @@
+/************************************************************
+ * <bsn.cl fy=2014 v=onl>
+ *
+ *        Copyright 2014, 2015 Big Switch Networks, Inc.
+ *        Copyright 2017 Delta Networks, Inc.
+ *
+ * Licensed under the Eclipse Public License, Version 1.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *        http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ *
+ * </bsn.cl>
+ ************************************************************
+ *
+ *
+ ***********************************************************/
+#include <onlp/platformi/sfpi.h>
+#include <x86_64_dni_wb2448/x86_64_dni_wb2448_config.h>
+#include "x86_64_dni_wb2448_log.h"
+
+#define SFP_PORT_BASE	49
+
+static int sfp_count__ = x86_64_dni_wb2448_CONFIG_SFP_COUNT;
```

Comment:
@vincent0083 We have this scenario on a few other platforms as well. 
The mechanism to implement this in ONL is for the process that runs the SDK to export well-known domain sockets that are then read by the ONLP implementation for that platform. 

This mechanism is also currently in place to implement the export of the BRCM switch thermal sensors to the ONLP API. 

The ONLP FILE APIs already support domain sockets transparently for this purpose. 
I have recently added a domain socket server for the SDK side to help facilitate management and registration on the server side of the socket. 

Implementing this option is simple and gives all ONLP clients access to the SFP data as long as the BRCM agent is running. 

I would be happy to provide you with example code of how this is implemented for other platforms with the same feature so you can implement the proper SFP support.